fix(receiver): TypeScript build errors in drizzle postgres adapters

## Problem

Vercel deploy shows TypeScript errors during build (non-blocking — deploy succeeds, but errors should be fixed):

```
apps/receiver/src/storage/drizzle/postgres.ts(80,42): error TS2345: Argument of type 'string | Sql<{}>' is not assignable to parameter of type 'string'.
  Type 'Sql<{}>'' is not assignable to type 'string'.
apps/receiver/src/telemetry/drizzle/postgres.ts(102,42): error TS2345: Argument of type 'string | Sql<{}>' is not assignable to parameter of type 'string'.
  Type 'Sql<{}>' is not assignable to type 'string'.
```

## Reproduction

```bash
cd /tmp && git clone https://github.com/muras3/3am.git
cd 3am && pnpm install && pnpm build
```

Or deploy to Vercel — errors appear in build logs.

## Root Cause

Drizzle ORM's `sql` template literal returns `Sql<{}>` but the code passes it where a plain `string` is expected. Likely needs a `.toString()` call or a type narrowing.

## Impact

- Build warnings on every Vercel deploy
- Could become blocking if `strict` tsconfig is enforced

--- a/apps/receiver/src/storage/drizzle/postgres.ts
+++ b/apps/receiver/src/storage/drizzle/postgres.ts
@@ -77,7 +77,8 @@ export class PostgresAdapter implements StorageDriver {
 
   constructor(connectionStringOrClient?: string | SharedPostgresClient) {
     if (typeof connectionStringOrClient === "string" || connectionStringOrClient === undefined) {
-      this.client = createPostgresClient(connectionStringOrClient);
+      const connStr = typeof connectionStringOrClient === "string" ? connectionStringOrClient : undefined;
+      this.client = createPostgresClient(connStr);
       this.ownsClient = true;
     } else {
       this.client = connectionStringOrClient;

--- a/apps/receiver/src/telemetry/drizzle/postgres.ts
+++ b/apps/receiver/src/telemetry/drizzle/postgres.ts
@@ -99,7 +99,8 @@ export class PostgresTelemetryAdapter implements TelemetryStoreDriver {
 
   constructor(connectionStringOrClient?: string | SharedPostgresClient) {
     if (typeof connectionStringOrClient === "string" || connectionStringOrClient === undefined) {
-      this.client = createPostgresClient(connectionStringOrClient);
+      const connStr = typeof connectionStringOrClient === "string" ? connectionStringOrClient : undefined;
+      this.client = createPostgresClient(connStr);
       this.ownsClient = true;
     } else {
       this.client = connectionStringOrClient;

